components/core/src/TimestampPattern.cpp

```diff
@@ -503,10 +521,56 @@ bool TimestampPattern::parse_timestamp (const string& line, epochtime_t& timesta
                     break;
                 }
 
+                case '#': {
+                    is_relative = true;
+                    break;
+                }
+
                 default:
                     return false;
             }
             is_specifier = false;
+        } else if (is_relative) {
+            int cFieldLength = 0;
+            // no leading zeroes currently supported for relative timestamp
+            if (line[line_ix] == '0') {
+                return false;
+            }
+            while (line_ix + cFieldLength < line_length) {
+                if ('0' <= line[line_ix + cFieldLength] &&
+                    line[line_ix + cFieldLength] <= '9') {
+                    cFieldLength++;
+                } else {
+                    break;
+                }
+            }
```

Comment:
```suggestion
            for (auto i = line_ix; i < line_length; ++i) {
                auto c = line[i];
                if (c < '0' || '9' < c) {
                    break;
                }
                ++field_length;
            }
```

Reply:
wouldn't it be this:
```
            for (auto i = line_ix + field_length; i < line_length; ++field_length) {
                auto c = line[i];
                if (c < '0' || '9' < c) {
                    break;
                }
            }
```

Reply:
Isn't field_length 0 before this loop?

Reply:
Oh yea, you're right, so it should be fine the way you had it.